renkuaqs/__init__.py

```diff
@@ -47,28 +51,59 @@ def _check_renku_version():
                      "You should consider install the suggested version.",)
 
 
-class GetGraphHandler(SimpleHTTPRequestHandler):
+# class RenkuFilesWatcher(LoggingEventHandler):
+#
+#     def on_modified(self, event):
+#         super().on_deleted(event)
+#
+#         if event.is_directory:
+#             content_dir = os.listdir(event.src_path)
+#             if len(content_dir) == 0:
+#                 logging.info(f"Directory {event.src_path} is now empty... graph can be refreshed")
+#             else:
+#                 logging.info(f"Directory {event.src_path} contains: {content_dir}")
+
+
+class HTTPGraphHandler(SimpleHTTPRequestHandler):
     def __init__(self, request, client_address, *args, **kwargs) -> None:
         super().__init__(request, client_address, *args, **kwargs)
         self.logger = logging.getLogger(self.__class__.__name__)
 
     def do_GET(self) -> None:
+        mount_path_env = os.environ.get('MOUNT_PATH', None)
+        logging.info(f'self.path = {self.path}, os.cwd = {os.getcwd()}, mount_path = {mount_path_env}')
         if self.path == '/':
-            aqsPlugin.build_graph(paths=os.getcwd(), template_location="remote")
-            if os.path.exists(os.path.join(os.getcwd(), 'graph.html')):
-                if 'MOUNT_PATH' in os.environ:
-                    self.path = os.path.join(os.environ['MOUNT_PATH'], 'graph.html')
-                else:
-                    self.path = 'graph.html'
+            try:
+                self.send_response(200)
+                self.send_header("Content-type", "text/html")
+                self.end_headers()
+                graph_html_content, ttl_content = graph_utils.build_graph_html(None, paths=os.getcwd(),
+                                                                               template_location="remote")
+                self.wfile.write(graph_html_content.encode())
+            except Exception as e:
+                output_html = f'''
+                <html><p>Error while generating the output graph:</p>
```

Comment:
at least head and body are probably useful

Reply:
added, also added some more formatting:

![image](https://user-images.githubusercontent.com/3757833/220316197-c04e45a3-e0fb-4586-95b6-2afefb833042.png)